azurelinuxagent/common/cgroups/cgroups.py

```diff
@@ -474,6 +340,30 @@ def _try_mkdir(path):
                 else:
                     raise
 
+    @staticmethod
+    def _try_rmdir(path):
```

Comment:
this functions seems to be used only from remove_cgroup(), which does not seem to be used

Reply:
It's used in the tests `TestExtensionWithCGroupsEnabled`. This is to cleanup the cgroup for tests.

```
class TestExtensionWithCGroupsEnabled(AgentTestCase):
    @staticmethod
    def _cleanup_cgroups(cgroup):
        try:
            cgroup.remove_cgroup()
        except CGroupsException:
            pass

        CGroupsTelemetry.stop_tracking(cgroup.name)
```

Reply:
should be moved to the test then?

Reply:
Its more useful for removing cgroups in general, and we should add that logic into CGroups. This is similar to the `stop_tracking` call done in `CGroupsTelemetry` class. It is specific to the cgroup.

Reply:
I'd remove if from the agent if it is not currently needed, we can always add it when needed. 

If you want to add it, then it needs improvements and unit tests (improvements - for example: both the name and documentation imply the function can be used to remove a directory, but the implementation includes some sort of error handling that seems specific to cgroups.

Reply:
Got it. Yes, refactoring/cleaning up the whole Cgroups is another big task. I'll move this method.